subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/LauncherServices.java

```diff
@@ -153,15 +154,22 @@ BuildTreeBuildActionExecutor createActionExecuter(List<BuildActionRunner> buildA
                                                           BuildStateRegistry buildStateRegistry,
                                                           PayloadSerializer payloadSerializer,
                                                           BuildOperationNotificationValve buildOperationNotificationValve,
-                                                          BuildCancellationToken buildCancellationToken
+                                                          BuildCancellationToken buildCancellationToken,
+                                                          ConfigurationCacheSupport configurationCacheSupport
         ) {
-            return new InProcessBuildActionExecuter(buildStateRegistry, payloadSerializer, buildOperationNotificationValve, buildCancellationToken,
+            return new InProcessBuildActionExecuter(
+                buildStateRegistry,
+                payloadSerializer,
+                buildOperationNotificationValve,
+                buildCancellationToken,
                 new RunAsBuildOperationBuildActionRunner(
                     new BuildCompletionNotifyingBuildActionRunner(
                         new FileSystemWatchingBuildActionRunner(
                             new ValidatingBuildActionRunner(
                                 new BuildOutcomeReportingBuildActionRunner(styledTextOutputFactory,
-                                    new ChainingBuildActionRunner(buildActionRunners)))))));
+                                    new ChainingBuildActionRunner(buildActionRunners)))))),
+                configurationCacheSupport
```

Comment:
💄 I'd move the `configurationCacheSupport` parameter to be after `buildCancellationToken`  for symmetry.